### ISSUE TITLE
Promise: Explore page

### ISSUE THREAD

**Issue by mixth** [wevisdemo/parliament-watch#97]
The PR shall be merged into `promise/main`.

This issue is blocked by:
- #96 

# Purpose
Explore page helps users explore any specific filtered promises more efficiently. There are four entry points for this page:
~~1. "All promises" button in main `/promises` page.~~
2. By any keyword
3. By any category
4. By any party

# Suggested Query Parameters
~~This page should list all promises if no query parameters is presented.~~
- keyword: string | undefined
- category: string | undefined
- party: string | undefined

# UI

Please refer to [the latest Figma](https://www.figma.com/design/ywiKyd0XIgDXf8crX4aoYE/Design_Phase1-2-%5BPromise-Tracker%5D?node-id=4949-11186&node-type=frame&t=loPbqUyXBGsvLv1C-0) by the design team.

![result_desktop](https://github.com/user-attachments/assets/6066e3e7-2275-40db-abf1-65a62870ffcc)

**Comment by Th1nkK1D** [comment]
Merged #109
